modules/core/src/v2/coins/cspr.ts

```diff
@@ -201,4 +222,63 @@ export class Cspr extends BaseCoin {
       .call(this)
       .asCallback(callback);
   }
+
+  /**
+   * Explain a Casper transaction from Raw Tx
+   *
+   * @param params
+   * @param callback
+   */
+  explainTransaction(
+    params: ExplainTransactionOptions,
+    callback?: NodeCallback<TransactionExplanation>
+  ): Bluebird<TransactionExplanation> {
+    const self = this;
+    return co<TransactionExplanation>(function*() {
+      const txHex = params.txHex || (params.halfSigned && params.halfSigned.txHex);
+      if (!txHex || !params.fee) {
+        throw new Error('missing explain tx parameters');
+      }
+      const txBuilder = accountLib.getBuilder(self.getChain()).from(txHex);
+
+      const tx: any = yield txBuilder.build();
+      if (!tx) {
+        throw new InvalidTransactionError('Error while trying to build transaction');
+      }
+      const id = Buffer.from(tx.casperTx.hash).toString('hex');
+      const transferId = accountLib.Cspr.Utils.getTransferId(tx.casperTx.session);
+      const amount = accountLib.Cspr.Utils.getTransferAmount(tx.casperTx.session);
+      const toAddress = accountLib.Cspr.Utils.getTransferDestinationAddress(tx._deploy.session);
+      const outputs: TransactionOutput[] = [];
+      // TODO(https://bitgoinc.atlassian.net/browse/STLX-677): define outputs for different types of tx
+      if (tx.type === accountLib.BaseCoin.TransactionType.Send) {
+        outputs.push({
+          address: toAddress,
+          amount,
+          coin: self.getChain(),
+        });
+      }
+      const outputAmount = outputs
+        .reduce((acumulator, output) => {
+          const currentValue = new BigNumber(output.amount);
+          return acumulator.plus(currentValue);
+        }, new BigNumber(0))
```

Comment:
Explain those magic numbers

Reply:
It's just the default accumulator value in the reduce. Here I'm defining the acumulator as a Big Number with value 0, e.g. if we don't have any outputs the returned value will be 0.